### DISCUSSION TITLE
getValues() returns an array for a single value

### DISCUSSION THREAD

**Discussion by hdaws** [react-hook-form/react-hook-form#8226] [Q&A] [answered]
This may be my misunderstanding of the documentation but the following sequence of events happens in my code and I do not understand why I receive an array of length 1 with the object I am looking for, instead of the actual object. This is a simplified example so my use case will not be obvious here. 

```
const initialValues = {
   field1: {
       propertyA: '',
       propertyB: 1,
       propertyC: []
   },
   field2:{
        propertyA: '',
       propertyB: 1,
       propertyC: []
       },
field3: [{...an array of other objects}]
}

  const methods = useForm({
    defaultValues: initialValues
  });
  
  let foundIt = methods.getValues('field1');
  
```
  Currently foundIt is an array that contains field1.
  The documentation leads me to believe it should *be* field 1.
  
  All of this is happening before I register my inputs so I was relying on the default values to let the form know what the contents are/get at them.

**Comment by JamesWong1999** [comment, marked as answer]
can you share a codesandbox?

**Comment by hdaws** [comment]
@JamesWong1999 
Hilariously, that provided my answer. in trying to create a simple codesandbox repo I found the error in my own code that was actually modifying my initial values to be an array (which shouldn't have happened).

So this is actually not a problem at all with react-hook-form. Many apologies for the trouble.  (I'll make sure to do that first next time before starting a discussion).
